src/core/core.controller.js

```diff
@@ -394,11 +394,41 @@ helpers.extend(Chart.prototype, /** @lends Chart */ {
 		scaleService.addScalesToLayout(this);
 	},
 
+	/**
+	 * @private
```

Comment:
It took me a little while to figure out exactly how this worked. Maybe add a JSDoc?
```
Updates the given metaset with the given index. Ensures it's stored at that index in the _metasets array by swapping with the metaset at that index if necessary.
@param {Object} meta the dataset metadata
@param {number} index the dataset index
```